zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -241,6 +296,8 @@ impl UpdateWith<PreparedBlock> for Chain {
             .expect("work has already been validated");
         self.partial_cumulative_work -= block_work;
 
+        // Note: the MMR is not reverted here; it's rebuilt and passed in fork()
+
```

Comment:
Here's an alternative way we could check this condition:
- make the `mmr` field optional
- `impl Clone` and set the `mmr` to `None`
- `assert!` that the `mmr` is `None` here
- `expect` that the `mmr` is `Some()` everywhere else

This design would prevent some MMR bugs, but it's also slightly harder to work with.
What do you think?

Reply:
I like the idea, but there is a catch: this function can be called either by `pop_tip` or `pop_root`. In `pop_root` the MMR will exist, so we can't assert that it's is `None` here.

I left it as is for now